mlfinlab/data_structures/standard_data_structures.py

```diff
@@ -0,0 +1,183 @@
+        else:
+            # Reset counters
+            cum_ticks, cum_dollar_value, cum_volume, high_price, low_price = 0, 0, 0, -np.inf, np.inf
+
+        return cum_ticks, cum_dollar_value, cum_volume, high_price, low_price
+
+    def _update_cache(self, date_time, price, low_price, high_price, cum_ticks, cum_volume, cum_dollar_value):
+        """
+        Update the cache which is used to create a continuous flow of bars from one batch to the next.
+
+        :param date_time: Timestamp of the bar
+        :param price: The current price
+        :param low_price: Lowest price in the period
+        :param high_price: Highest price in the period
+        :param cum_ticks: Cumulative number of ticks
+        :param cum_volume: Cumulative volume
+        :param cum_dollar_value: Cumulative dollar value
+        """
+        cache_data = self.cache_tuple(
+            date_time, price, high_price, low_price, cum_ticks, cum_volume, cum_dollar_value)
+        self.cache.append(cache_data)
+
+
+def get_dollar_bars(file_path, threshold=70000000, batch_size=20000000, verbose=True, to_csv=False):
+    """
+    Creates the dollar bars: date_time, open, high, low, close.
+
+    Following the paper "The Volume Clock: Insights into the high frequency paradigm" by Lopez de Prado, et al,
+    it is suggested that using 1/50 of the average daily dollar value, would result in more desirable statistical
+    properties.
+
+    :param file_path: File path pointing to csv data.
+    :param threshold: A cumulative value above this threshold triggers a sample to be taken.
+    :param batch_size: The number of rows per batch. Less RAM = smaller batch size.
+    :param verbose: Print out batch numbers (True or False)
+    :param to_csv: Save bars to csv after every batch run (True or False)
+    :return: Dataframe of dollar bars
+    """
+
+    bars = StandardBars(file_path=file_path, metric='cum_dollar_value',
+                        threshold=threshold, batch_size=batch_size)
+    dollar_bars = bars.batch_run(verbose=verbose, to_csv=to_csv)
+    return dollar_bars
+
+
+def get_volume_bars(file_path, threshold=28224, batch_size=20000000, verbose=True, to_csv=False):
+    """
+    Creates the volume bars: date_time, open, high, low, close.
+
+    Following the paper "The Volume Clock: Insights into the high frequency paradigm" by Lopez de Prado, et al,
+    it is suggested that using 1/50 of the average daily volume, would result in more desirable statistical properties.
+
+    :param file_path: File path pointing to csv data.
+    :param threshold: A cumulative value above this threshold triggers a sample to be taken.
+    :param batch_size: The number of rows per batch. Less RAM = smaller batch size.
+    :param verbose: Print out batch numbers (True or False)
+    :param to_csv: Save bars to csv after every batch run (True or False)
+    :return: Dataframe of volume bars
+    """
+    bars = StandardBars(file_path, 'cum_volume', threshold, batch_size)
```

Comment:
Why does the prev function specify which these areguements are and this one doesnt?